internal/http_api/api_request.go

```diff
@@ -28,15 +28,17 @@ func (c *deadlinedConn) Write(b []byte) (n int, err error) {
 
 // A custom http.Transport with support for deadline timeouts
 func NewDeadlineTransport(connectTimeout time.Duration, requestTimeout time.Duration) *http.Transport {
+	// arbitrary values copied from http.DefaultTransport
 	transport := &http.Transport{
-		Dial: func(netw, addr string) (net.Conn, error) {
-			c, err := net.DialTimeout(netw, addr, connectTimeout)
-			if err != nil {
-				return nil, err
-			}
-			return &deadlinedConn{connectTimeout, c}, nil
```

Comment:
can we drop the `deadlinedConn` struct/methods above?

Reply:
nice catch, silly of me to leave that stuff